Mathlib/Algebra/Algebra/Operations.lean

```diff
@@ -0,0 +1,750 @@
+      ext S
+      constructor <;> rintro ⟨y, hy⟩
+      · use ⟨f y, mem_map.mpr ⟨y.1, y.2, rfl⟩⟩  -- porting note: added `⟨⟩`
+        refine' Eq.trans _ hy
+        ext
+        simp
+      · obtain ⟨y', hy', fy_eq⟩ := mem_map.mp y.2
+        use ⟨y', hy'⟩  -- porting note: added `⟨⟩`
+        refine' Eq.trans _ hy
+        rw [f.toLinearMap_apply] at fy_eq
+        ext
+        simp [fy_eq]
+
+#align submodule.map_mul Submodule.map_mul
+
+theorem map_op_mul :
+    map (↑(opLinearEquiv R : A ≃ₗ[R] Aᵐᵒᵖ) : A →ₗ[R] Aᵐᵒᵖ) (M * N) =
+      map (↑(opLinearEquiv R : A ≃ₗ[R] Aᵐᵒᵖ) : A →ₗ[R] Aᵐᵒᵖ) N *
+        map (↑(opLinearEquiv R : A ≃ₗ[R] Aᵐᵒᵖ) : A →ₗ[R] Aᵐᵒᵖ) M := by
+  apply le_antisymm
+  · simp_rw [map_le_iff_le_comap]
+    refine' mul_le.2 fun m hm n hn => _
+    rw [mem_comap, map_equiv_eq_comap_symm, map_equiv_eq_comap_symm]
+    show op n * op m ∈ _
+    exact mul_mem_mul hn hm
+  · refine' mul_le.2 (MulOpposite.rec' fun m hm => MulOpposite.rec' fun n hn => _)
+    rw [Submodule.mem_map_equiv] at hm hn⊢
+    exact mul_mem_mul hn hm
+#align submodule.map_op_mul Submodule.map_op_mul
+
+theorem comap_unop_mul :
+    comap (↑(opLinearEquiv R : A ≃ₗ[R] Aᵐᵒᵖ).symm : Aᵐᵒᵖ →ₗ[R] A) (M * N) =
+      comap (↑(opLinearEquiv R : A ≃ₗ[R] Aᵐᵒᵖ).symm : Aᵐᵒᵖ →ₗ[R] A) N *
+        comap (↑(opLinearEquiv R : A ≃ₗ[R] Aᵐᵒᵖ).symm : Aᵐᵒᵖ →ₗ[R] A) M :=
+  by simp_rw [← map_equiv_eq_comap_symm, map_op_mul]
+#align submodule.comap_unop_mul Submodule.comap_unop_mul
+
+theorem map_unop_mul (M N : Submodule R Aᵐᵒᵖ) :
+    map (↑(opLinearEquiv R : A ≃ₗ[R] Aᵐᵒᵖ).symm : Aᵐᵒᵖ →ₗ[R] A) (M * N) =
+      map (↑(opLinearEquiv R : A ≃ₗ[R] Aᵐᵒᵖ).symm : Aᵐᵒᵖ →ₗ[R] A) N *
+        map (↑(opLinearEquiv R : A ≃ₗ[R] Aᵐᵒᵖ).symm : Aᵐᵒᵖ →ₗ[R] A) M :=
+  have : Function.Injective (↑(opLinearEquiv R : A ≃ₗ[R] Aᵐᵒᵖ) : A →ₗ[R] Aᵐᵒᵖ) :=
+    LinearEquiv.injective _
+  map_injective_of_injective this <| by
+    rw [← map_comp, map_op_mul, ← map_comp, ← map_comp, LinearEquiv.comp_coe,
+      LinearEquiv.symm_trans_self, LinearEquiv.refl_toLinearMap, map_id, map_id, map_id]
+#align submodule.map_unop_mul Submodule.map_unop_mul
+
+theorem comap_op_mul (M N : Submodule R Aᵐᵒᵖ) :
+    comap (↑(opLinearEquiv R : A ≃ₗ[R] Aᵐᵒᵖ) : A →ₗ[R] Aᵐᵒᵖ) (M * N) =
+      comap (↑(opLinearEquiv R : A ≃ₗ[R] Aᵐᵒᵖ) : A →ₗ[R] Aᵐᵒᵖ) N *
+        comap (↑(opLinearEquiv R : A ≃ₗ[R] Aᵐᵒᵖ) : A →ₗ[R] Aᵐᵒᵖ) M :=
+  by simp_rw [comap_equiv_eq_map_symm, map_unop_mul]
+#align submodule.comap_op_mul Submodule.comap_op_mul
+
+section
+
+open Pointwise
+
+-- TODO: lean4#2074
+set_option synthInstance.etaExperiment true in
+/-- `Submodule.pointwiseNeg` distributes over multiplication.
+
+This is available as an instance in the `pointwise` locale. -/
+protected def hasDistribPointwiseNeg {A} [Ring A] [Algebra R A] : HasDistribNeg (Submodule R A) :=
+  toAddSubmonoid_injective.hasDistribNeg _ neg_toAddSubmonoid mul_toAddSubmonoid
+#align submodule.has_distrib_pointwise_neg Submodule.hasDistribPointwiseNeg
```

Comment:
I don't know if there's a better workaround here

Reply:
Fixed, for now